Fix names and mark-ups of internal flags

muzimuzhi opened the pull request.

car222222 commented:
Not quite sure in all cases, exactly what is going on here!  

But "LGTM", as they write!

car222222 reviewed (approved):
See comment.

muzimuzhi commented:
Rebased to latest `main`, and
- corrected commit message about `@@_missing`
- changed to `variable` env rather than `macro` to mark-up flags
- catch flags created by `\flag_clear_new:n` in `l3str-convert.dtx`

@car222222 

The first commit in this PR corrects a problem introduced in https://github.com/muzimuzhi/latex3/commit/af69d11867215e32926197ce8b84c1dd8dbabfe8.

- Before af69d11867215e32926197ce8b84c1dd8dbabfe8, at the beginning of l3flag module doc, the introduction to flag data type contained
  > Flag variables are always local. They are referenced by a `⟨flag name⟩` such as `str_missing`.
- After af69d11867215e32926197ce8b84c1dd8dbabfe8, `str_missing` became `@@_missing`
  `@@` in `\texttt{@@_missing}` is not replaced by DocStrip, even replaced the result flag name `__flag_missing` is non-existing.
- This commit replaces `@@_missing` with an (other) existing public flag name `fp_overflow`.

The second commit,
- Before
  ![image](https://github.com/latex3/latex3/assets/6376638/48e23749-1ad8-4690-811f-87561d0b1a77)
  Note the prefix `flag␣` is not part of the flag name accessible to users, but rather represents implementation details that flag functions like `\flag_new:n {<flag name>}` internally use `\cs{flag <flag name>}` to hold info about that flag.
- After
  ![image](https://github.com/latex3/latex3/assets/6376638/e52bfc8f-ec52-4305-9f34-bde4af35152a)

The third commit simply corrects wrong and adds missing mark-ups for internal flags used by `l3kernel`.
  ```
  rg '\\flag_(clear_)?new:n\s*\{\s*@@' -g '*.dtx' --no-heading
  l3regex.dtx:3527:\flag_new:n { @@_cs }
  l3regex.dtx:6735:\flag_new:n { @@_begin }
  l3regex.dtx:6736:\flag_new:n { @@_end }
  l3str-convert.dtx:351:\flag_new:n { @@_byte }
  l3str-convert.dtx:352:\flag_new:n { @@_error }
  l3str-convert.dtx:1849:\flag_clear_new:n { @@_missing }
  l3str-convert.dtx:1850:\flag_clear_new:n { @@_extra }
  l3str-convert.dtx:1851:\flag_clear_new:n { @@_overlong }
  l3str-convert.dtx:1852:\flag_clear_new:n { @@_overflow }
  l3str-convert.dtx:2165:  \flag_clear_new:n { @@_missing }
  l3str-convert.dtx:2166:  \flag_clear_new:n { @@_extra }
  l3str-convert.dtx:2167:  \flag_clear_new:n { @@_end }
  l3str-convert.dtx:2466:  \flag_clear_new:n { @@_overflow }
  l3str-convert.dtx:2467:  \flag_clear_new:n { @@_end }
  l3fp-symbolic.dtx:618:\flag_new:n { @@_symbolic }
  ```
